core/arch/arm/mm/tee_mm.c

```diff
@@ -190,13 +196,13 @@ tee_mm_entry_t *tee_mm_alloc(tee_mm_pool_t *pool, size_t size)
 
 			if (remaining < size)
 				/* out of memory */
-				return NULL;
+				goto out;
 		}
 	}
 
 	nn = tee_mm_add(entry);
```

Comment:
Here's we're calling `malloc()` with a spinlock held, this more or less forces `malloc()` to also use spinlock even if that happens to be the case currently.
I'd rather allocate the new entry in advance before taking the spinlock.